sklearn/metrics/_classification.py

```diff
@@ -2629,8 +2630,10 @@ def log_loss(
             )
 
     # Renormalize
-    y_pred /= y_pred.sum(axis=1)[:, np.newaxis]
-    loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)
+    y_pred_sum = y_pred.sum(axis=1)
+    if np.any(np.abs(y_pred_sum - 1) >= 4 * np.finfo(float).eps):
```

Comment:
What's the reason for not always dividing by `y_pred_sum`?

Reply:
I thought about numerical rounding effects, but that may not be needed. I'll remove it.

BTW, I was very surprised when I found out this normalization step. Do you know the reason/use case for it? Strictly speaking, it gives wrong results.

Reply:
Why are you saying it gives wrong results? `y_pred` are probabilities that must sum to one, to it does seem reasonable to me to enforce that constraint in some way.

Reply:
It is reasonable to have sum(y_pred)=1, but bad models may do otherwise. IMHO, any such assumption or correction should not happen in the metric because it gives wrong (usually too good) results for such bad models.

Reply:
I feel like computing a log loss with probabilities not summing to one does not make sense, so I am ok with this correction.